include/pybind11/detail/common.h

```diff
@@ -162,6 +162,9 @@
 #include <memory>
 #include <typeindex>
 #include <type_traits>
+#if defined(__has_include) && __has_include(<version>)
+#    include <version>
+#endif
```

Comment:
Since C++20, `__cpp_lib_foo` feature test macros are in `<version>` header and at least clang respects that.